core/client/fs/src/main/java/alluxio/client/block/stream/DefaultBlockWorkerClient.java

```diff
@@ -248,4 +259,53 @@ public void freeWorker() {
   public ListenableFuture<LoadResponse> load(LoadRequest request) {
     return mRpcFutureStub.load(request);
   }
+
+  @Override
+  public ListenableFuture<GetBlockChecksumResponse> getBlockChecksum(
+      GetBlockChecksumRequest request) {
+    // Non-batched mode
+    if (request.getBlockIdsCount() < mChecksumBatchSize) {
+      long startTs = CommonUtils.getCurrentMs();
```

Comment:
Does the client just send requests without doing logical processing? Would it be better to put this part of the logic processing on a higher level?

Reply:
i moved the logic to the util